ui/index.js

```diff
@@ -52,18 +57,39 @@ export const updateBackgroundConnection = (backgroundConnection) => {
 };
 
 export default function launchMetamaskUi(opts, cb) {
+  console.log('launching');
   const { backgroundConnection } = opts;
-  // check if we are unlocked first
-  backgroundConnection.getState(function (err, metamaskState) {
-    if (err) {
-      cb(err, metamaskState);
-      return;
-    }
-    startApp(metamaskState, backgroundConnection, opts).then((store) => {
-      setupDebuggingHelpers(store);
-      cb(null, store);
+  browser.storage.local
+    .get(null)
+    .then((state) => {
+      if (!state[STORE_KEY]) {
+        throw new Error('No state yet');
+      }
+      console.log(state);
+      startApp(state[STORE_KEY], backgroundConnection, opts).then((store) => {
+        setupDebuggingHelpers(store);
+        cb(null, store);
+      });
+    })
+    .catch((error) => {
+      console.log('in the catch');
+      console.error(error);
+      // check if we are unlocked first
+      backgroundConnection.getState(function (err, metamaskState) {
```

Comment:
If we don't have the state in storage we fall back to asking for the background connection. The reason why this isn't a viable solution (caching locally instead of making the storage the DEFAULT) is because we have to have a copy of state in order to reliably start up without the background connection. If that key is missing we need the background connection, but without waiting for the service worker we do not have a reference to the streams to be able to request state and therefore this will fail and crash the application. We could check in the app/scripts/ui.js file if we have state locally prior to booting up the application, but a better solution is to make the local storage the default way we store and access state. Updates are written to local storage from the background and the UI just reads it. We can even subscribe to changes to this store in the UI instead of relying upon the message from the service worker that state has updated.